input/input_driver.c

```diff
@@ -3400,6 +3548,39 @@ bool input_keyboard_line_event(
    const char            *word = NULL;
    char            c           = (character >= 128) ? '?' : character;
 
+#ifdef HAVE_LANGEXTRA
+   static uint32_t composition = 0;		
+   if( state->size==0 )  composition = 0;
+   if( character && character<0xff) composition = 0;
+   composition = composition & 0xffffff;
+   if( character & 0xff000000)
```

Comment:
When is this conditional taken? Does it go into this conditional block when the characters are Korean, or what is the flag in this case?

Maybe it'd help readability if these masks were a macro define that tells us what they stand for, like if it's for korean characters or whatever.

Reply:
This applies to most 3byte-utf8 languages, as well as Korean. I haven't checked, but 2byte-utf8 (ex-latin?) will work.

Reply:
OK, so there is no conflict with Japanese language support here right, since that was already previously working? Or did Japanese text input not work at all before?

Reply:
Retroarch built in Japanese, still works. However, in previous version, all extended languages entered on the keyboard were replaced with "?".
in input_driver.c,  input_event_osk_append() allow 2 or 3 byte character. that's why built in japanese works before. but
in input_keyboard_line_event(), at first line,  you can see.
   char c = (character >= 128) ? '?' : character;     --> replace with '?' all extended character.
and not only that, it check macro ISPRINT()
   #define ISPRINT(c) (((int)(c) >= ' ' && (int)(c) <= '~') ? 1 : 0)   -->> means allow only 32(space) to 126(~)
At first version, I replace  "else if (ISPRINT(c))" to "else if (character)" to pass all extended character. and call input_event_osk_append() instead attaching 1 byte.
But this version, I attached bit 0xff000000 for 2-3 byte utf8, to saparate process from previous code. 
I guess, current version may works on japanese and chinese keyboard input too, in win32 and linux.  but may not works in other platform,
I attached 0xff000000 only in "win32_commonn.c" and do not process 2-3 byte without "0xff00000" in input_keyboard_line_event()